### ISSUE TITLE
[SUGGESTION] Add extra radar functionality, and have it update every tick

### ISSUE THREAD

**Issue by Alexandre425** [Stooberton/ACF-3#194]
A few suggestions to improve radar.

- Have radars consume power (once a way to generate it is implemented).
- Radars have two modes of operation, normal (air tracking) and millimeter wave (ground tracking) modes.

**In normal mode:**
- In normal mode, targets are acquired at the same rate they do now (for performance's sake) but update their position every tick.
- Radars check for the proximity of targets to solid background, giving more inaccurate position results the bigger the radar to target / target to background ratio. At a certain threshold, the target cannot be tracked at all.
- Radar notching is possible (maybe? not sure if it'll be meaningful in ACF combat, and would complicate things for stationary AA trying to hit aircraft flying perpendicular).

**In millimeter mode:**
- Targets are acquired at the same rate, but the position update is slower, once every 90-160ms.
- Consumes considerably more power.
- Proximity to background does not matter, output has no noise.

In short, tracking air targets becomes easier, and tracking ground targets perfectly becomes slightly harder, and you can't do both at the same time reliably.

**Comment by arthritisguy** [comment]
They could be linked to batteries from Electric Motors. Maybe the normal tracking mode could be dependent on combat mode, say in sandbox it would be perfectly accurate, in Classic it would be less accurate, and in Competitive it could be much less accurate. A way to generate power could be linking a petrol or Diesel engine to an Electric motor to produce power?
